modules/dns_nsupdate/dns_nsupdate_main.rb

```diff
@@ -13,22 +13,41 @@ def initialize(a_server = nil, a_ttl = nil)
     end
 
     def create_a_record(fqdn, ip)
-      do_create(fqdn, ip, "A")
+      case a_record_conflicts(fqdn, ip) #returns -1, 0, 1
+      when 1
+        raise(Proxy::Dns::Collision, "'#{fqdn} 'is already in use")
+      when 0 then
+        return nil
+      else
+        do_create(fqdn, ip, "A")
+      end
     end
 
-    def create_ptr_record(fqdn, ip)
-      do_create(ip, fqdn, "PTR")
+    def create_aaaa_record(fqdn, ip)
+      case aaaa_record_conflicts(fqdn, ip) #returns -1, 0, 1
+      when 1
+        raise(Proxy::Dns::Collision, "'#{fqdn} 'is already in use")
+      when 0 then
+        return nil
+      else
+        do_create(fqdn, ip, "AAAA")
+      end
```

Comment:
You might also want to explicitly return `nil` and alter the tests to use `asstert_nil` make sure not to break foreman